tracing/tracing.go

```diff
@@ -55,16 +55,30 @@ func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, err
 
 // ExtractTraceID extracts the trace id, if any from the context.
 func ExtractTraceID(ctx context.Context) (string, bool) {
+	if sctx := extractJaegerContext(ctx); sctx != nil {
+		return sctx.TraceID().String(), true
+	}
+	return "", false
+}
+
+// ExtractTraceSpanID extracts the trace id, span id if any from the context.
+func ExtractTraceSpanID(ctx context.Context) (string, string, bool) {
+	if sctx := extractJaegerContext(ctx); sctx != nil {
+		return sctx.TraceID().String(), sctx.SpanID().String(), true
+	}
+	return "", "", false
+}
+
+func extractJaegerContext(ctx context.Context) *jaeger.SpanContext {
 	sp := opentracing.SpanFromContext(ctx)
 	if sp == nil {
-		return "", false
+		return nil
 	}
 	sctx, ok := sp.Context().(jaeger.SpanContext)
 	if !ok {
-		return "", false
+		return nil
 	}
-
-	return sctx.TraceID().String(), true
+	return &sctx
```

Comment:
By returning `&sctx`, `jaeger.SpanContext` needs to be allocated on a heap first, and then pointer to it returned, right?

Perhaps we could return just traceID and spanID directly? Or `jaeger.SpanContext` without the pointer.

Reply:
I thought about that , but that would be duplicated code. I can definitely do a copy of the jeager.SpanContext thought, if that's preferred?

Reply:
By the way , I don't know if the compiler is smart enough to avoid allocation on the heap

Reply:
Can we keep ExtractTraceID, and call ExtractTraceSpanID inside, just discard spanID when return result. In that case no need to refectory this extractJaegerContext but leave all logics in ExtractTraceSpanID

Reply:
facepalm, of course

Reply:
I think it is a fair comment, current code LGTM, approved.

Reply:
it did inspire me to go ahead and look at the assembly :) Got the final result by deferring the formatting to the public function , but only returning the two relevant fields from the helper function. It's the least extra instructions.